src/main/java/org/opentosca/csarrepo/rest/resource/CsarFileResource.java

```diff
@@ -99,11 +97,22 @@ public Response getFile() {
 			}
 		}
 
-		File file = fs.getFile(matchedCsarFile.getHashedFile().getFilename());
-		String csarName = matchedCsarFile.getCsar().getName();
+		// TODO: check if it would be better if getFile gets a String
+		// TODO: use real UserID
+		DownloadCsarFileService downService = new DownloadCsarFileService(0L, matchedCsarFile.getId());
+
+		if (downService.hasErrors()) {
+			return Response.serverError().entity("DownloadService has Errors: " + downService.getErrors().get(0))
+					.build();
+		}
+
+		DownloadCsarFileObject csarFileObject = downService.getResult();
+
+		File file = csarFileObject.getFile();
 		ResponseBuilder response = Response.ok((Object) file);
-		response.header("Content-Disposition", "attachment; filename=" + csarName + ".csar");
+		// TODO: discuss if this is the correct fileName
+		response.header("Content-Disposition", "attachment; filename=" + matchedCsarFile.getName());
```

Comment:
the REST Service returns a zip named after the csarFile <csarFile.getName()>.csar

is this ..
a) the same way the servlet behaves
b) the correct way :smile: ?